include/zephyr/kernel_structs.h

```diff
@@ -257,6 +257,8 @@ __attribute_const__ struct k_thread *z_smp_current_get(void);
 #define _current _kernel.cpus[0].current
 #endif
 
+#define CPU_ID (CONFIG_MP_MAX_NUM_CPUS == 1) ? 0 : _current_cpu->id
```

Comment:
Purely for the `NO_OPTIMIZATIONS`  case I think it would make sense to split the cases by `#if`.
If not, it should at least be surrounded by brackets.